src/invidious/routes/before_all.cr

```diff
@@ -100,6 +88,24 @@ module Invidious::Routes::BeforeAll
       end
     end
 
+    unregistered_path_whitelist = {"/", "/login", "/licenses", "/privacy"}
+    if CONFIG.login_only && !env.get?("user") && !unregistered_path_whitelist.includes?(env.request.path)
```

Comment:
```suggestion
    unregistered_path_whitelist = {"/api/v1/comments/jNQXAC9IVRw", "/api/v1/videos/dQw4w9WgXcQ", "/feed/webhook/v1:", "/login", "/licenses", "/opensearch.xml", "/privacy"}
    if CONFIG.login_only && !env.get?("user") && !unregistered_path_whitelist.any? { |r| env.request.path.starts_with? r }
```

Changing it to this comes with the advantage, that the login page is opened first. It also eliminates some `302` and lets the health-check succeed. Furthermore, it allows adding the instance and logging in with an account in third-party apps like Yattee.

Reply:
This is good, but I also found that Clipious client requires `/api/v1/stat` when adding a new server to the client.  It loads this to check it is a valid server, and refuses if it doesn't work (currently it returns a 500 error, which should be a bug anyway).  However, even if this additional endpoint is added the client doesn't load Popular and Trending (default pane) because it makes these requests unauthenticated (will need to raise this as an issue with Clipious)

Reply:
Added your input to a new suggestion.